website/src/components/TutorialComponents/CodeBlock.tsx

```diff
@@ -36,13 +38,24 @@ export function GithubDiscussionBtn({ repo }) {
 // source code of LiveEditor that allows for code editing
 // a good starting point for customizing our own code editor
 
-const CodeEditor = (props) => {
+function CodeEditor(props: {
+    className: string,
+    code: string,
+    disabled: boolean,
+    language: string,
+    onChange: (code: string) => void,
+    prism: typeof Prism,
+    style?: CSSProperties,
+    theme: PrismTheme,
+    githubRepo: string | undefined,
+    showLineNumbers: boolean,
+}) {
     const editorRef = useRef(null);
     const [code, setCode] = useState(props.code || "");
 
     useEffect(() => {
-        setCode(props.code);
-    }, [props.code]);
+        setCode(code);
+    }, [code]);
```

Comment:
What is the purpose of this effect?

Reply:
Wrong edits. Should be `props.code` not the local variable `code`.

Reply:
The side effect set the value of `code` variable to `props.code` again if `props.code` changes.